pkg/broker/broker.go

```diff
@@ -74,45 +74,67 @@ func (b *Broker) Provision(request *osb.ProvisionRequest, c *broker.RequestConte
 		return nil, err
 	}
 
-	err := b.Client.Provision(request.InstanceID, request.ServiceID, request.PlanID, namespace, request.Parameters)
+	glog.V(5).Infof("Provisioning %s (%s/%s) in %s", request.InstanceID, request.ServiceID, request.PlanID, namespace)
+
+	operationName, err := b.Client.Provision(request.InstanceID, request.ServiceID, request.PlanID, namespace, request.AcceptsIncomplete, request.Parameters)
 	if err != nil {
 		glog.Errorln(err)
 		return nil, err
 	}
 
 	response := broker.ProvisionResponse{}
-	if request.AcceptsIncomplete {
-		response.Async = b.async
+	if request.AcceptsIncomplete && operationName != "" {
```

Comment:
Hmm, actually, that should never happen.
Error on line 79 is handled in line 80~82 (early return).
`operationName` can only be empty if the operation was done synchronously. But in that case `request.AcceptsIncomplete` would be false.

I can put a `panic` here instead, if you'd prefer (because it would mean program flow went somewhere really whacky, basically a run-time assert)?

Reply:
Nope, that's fine (I really don't want a panic) I just wasn't sure if there was a hidden control flow in here or not.

How would you feel about simply using `if acceptsIncomplete` without the check against operation name is empty? Otherwise I think taking what you just wrote and putting it into a comment would help clarify why that if statement looks the way it does would help people who read it in the future. I prefer the first fix (remove the second condition) but the comment works too.

Reply:
Fixed in b4b5d0124ec1775d8209d36066b3c1d226f07359